launchlib/monitor.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package launchlib
+
+import (
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	CheckPeriod = 5 * time.Second
+)
+
+type ProcessMonitor struct {
+	ServicePid     int
```

Comment:
Would prefer not to bake in assumptions about 'service' - can we just call these a PrimaryPid and ProcessGroupID ?